core/src/main/kotlin/xtdb/api/YamlSerde.kt

```diff
@@ -14,10 +20,59 @@ import xtdb.api.log.InMemoryLogFactory
 import xtdb.api.log.LocalLogFactory
 import xtdb.api.log.LogFactory
 import xtdb.api.storage.ObjectStoreFactory
-import java.util.ServiceLoader
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.*
 import java.util.ServiceLoader.Provider
 import kotlin.reflect.KClass
 
+object EnvironmentVariableProvider {
+    fun getEnvVariable(name: String): String? = System.getenv(name)
+}
+
+fun envFromTaggedNode(taggedNode: YamlTaggedNode ): String {
+    if (taggedNode.tag == "!Env") {
+        val value = taggedNode.innerNode.yamlScalar.content
+        return EnvironmentVariableProvider.getEnvVariable(value) ?: throw IllegalArgumentException("Environment variable '$value' not found")
+    }
+    return taggedNode.innerNode.yamlScalar.content
+}
+fun handleEnvTag(input: YamlInput): String {
+    val currentLocation = input.getCurrentLocation()
+    val scalar = input.node
```

Comment:
Seems to work at least 2 levels deep in the map, so presumably it only gives you the outer object surrounding the scalar. Not entirely sure.